solr/core/src/test/org/apache/solr/core/TestLazyCores.java

```diff
@@ -212,43 +225,44 @@ public void testCachingLimit() throws Exception {
       checkCoresNotLoaded(cc, "collection3", "collection4", "collection6", "collection7", "collection8", "collection9");
 
       // By putting these in non-alpha order, we're also checking that we're  not just seeing an artifact.
-      SolrCore core1 = cc.getCore("collection1");
-      SolrCore core3 = cc.getCore("collection3");
-      SolrCore core4 = cc.getCore("collection4");
-      SolrCore core2 = cc.getCore("collection2");
-      SolrCore core5 = cc.getCore("collection5");
+      getCoreAndPutBack(cc, "collection1");
+      getCoreAndPutBack(cc, "collection3");
+      getCoreAndPutBack(cc, "collection4");
+      getCoreAndPutBack(cc, "collection2");
+      getCoreAndPutBack(cc, "collection5");
 
       checkLoadedCores(cc, "collection1", "collection2", "collection3", "collection4", "collection5");
       checkCoresNotLoaded(cc, "collection6", "collection7", "collection8", "collection9");
 
       // map should be full up, add one more and verify
-      SolrCore core6 = cc.getCore("collection6");
+      getCoreAndPutBack(cc, "collection6");
       checkLoadedCores(cc, "collection1", "collection2", "collection3", "collection4", "collection5",
           "collection6");
       checkCoresNotLoaded(cc, "collection7", "collection8", "collection9");
 
-      SolrCore core7 = cc.getCore("collection7");
+      getCoreAndPutBack(cc, "collection7");
       checkLoadedCores(cc, "collection1", "collection2", "collection3", "collection4", "collection5",
           "collection6", "collection7");
       checkCoresNotLoaded(cc, "collection8", "collection9");
 
-      SolrCore core8 = cc.getCore("collection8");
+      getCoreAndPutBack(cc, "collection8");
       checkLoadedCores(cc, "collection1", "collection4", "collection5", "collection8");
```

Comment:
Simply because collection1, colllection4 and collection5 are not transient so they are not in the transient core cache with limited size. See the testCores constant above (isTransient is the 2nd param - thanks IntelliJ for displaying the param name).

Reply:
Thanks @bruno-roustant. Giving different names to the collections to reflect their properties would have saved me from asking a stupid question and would make the test easier to read/understand (I know this is ooooold code).